src/distances/alignment_helpers.rs

```diff
@@ -0,0 +1,559 @@
+    }
+
+    // Set values for the body of the table
+    for row in 1..(len_y + 1) {
+        for col in 1..(len_x + 1) {
+            // Check if sequences match at position col-1 in x and row-1 in y
+            // Reason for subtraction is that NW considers an artificial gap at the start
+            // of each sequence, so the dp tables' indices are 1 higher than that of
+            // the actual sequences
+            let mismatch_penalty = if x[col - 1] == y[row - 1] { 0 } else { 1 };
+            let new_cell = [
+                table[row - 1][col - 1].0 + mismatch_penalty,
+                table[row - 1][col].0 + gap_penalty,
+                table[row][col - 1].0 + gap_penalty,
+            ]
+            .into_iter()
+            .zip([Direction::Diagonal, Direction::Up, Direction::Left].into_iter())
+            .min_by(|x, y| x.0.cmp(&y.0))
+            .unwrap();
+
+            table[row][col] = new_cell;
+        }
+    }
+
+    table
+}
+
+pub enum Edit<T: Number> {
+    Deletion(usize),
+    Insertion(usize, T),
+    Substitution(usize, T),
+}
+
+// Given an alignement of two sequences, returns the set of edits needed to turn sequence
+// x into sequence y
+pub fn alignment_to_edits<T: Number>(aligned_x: &Vec<T>, aligned_y: &Vec<T>) -> Vec<Edit<T>> {
+    aligned_x
+        .iter()
+        .zip(aligned_y.iter())
+        .filter(|(x, y)| x != y)
+        .enumerate()
+        .map(|(index, (x, y))| {
+            if *x == T::from(b'-').unwrap() {
+                Edit::Insertion(index, *y)
+            } else if *y == T::from(b'-').unwrap() {
+                Edit::Deletion(index)
+            } else {
+                Edit::Substitution(index, *y)
+            }
+        })
+        .collect()
+}
+
+// Iterative version of traceback function so we can benchmark both this and recursive option
+pub fn traceback_iterative<T: Number>(
+    table: &Vec<Vec<(usize, Direction)>>,
+    unaligned_seqs: (&[T], &[T]),
+) -> (Vec<T>, Vec<T>) {
+    let mut row_index = table.len() - 1;
+    let mut column_index = table[0].len() - 1;
+
+    let (mut aligned_x, mut aligned_y) = (Vec::<T>::new(), Vec::<T>::new());
+    let (unaligned_x, unaligned_y) = unaligned_seqs;
+    let mut direction = table[row_index][column_index].1;
+
+    while direction != Direction::None {
+        match direction {
+            Direction::Diagonal => {
+                aligned_x.insert(0, unaligned_x[column_index - 1]);
+                aligned_y.insert(0, unaligned_y[row_index - 1]);
+                row_index -= 1;
+                column_index -= 1;
+            }
+            Direction::Left => {
+                aligned_x.insert(0, unaligned_x[column_index - 1]);
+                aligned_y.insert(0, T::from(b'-').unwrap());
+                column_index -= 1;
+            }
+            Direction::Up => {
+                aligned_x.insert(0, T::from(b'-').unwrap());
+                aligned_y.insert(0, unaligned_y[row_index - 1]);
```

Comment:
Instead of `insert`ing at index `0`, push to the end of the vectors. Then, when returning from the function, reverse the vectors.

Reply:
Sounds good